### DISCUSSION TITLE
arbitrarily redirect user after auth?

### DISCUSSION THREAD

**Discussion by tconroy** [supabase/supabase#4592] [Questions]
If a user visits `mydomain.com/app/route-that-needs-auth`, and they are not logged in, I want to redirect them to the signin screen, `/signin`.

Once they authenticate (ie via Oauth/SSO, email, magic link) or register, I then went to redirect them back to their original route (in this example, `/app/route-that-needs-auth`).

Is this possible?

I've tried adding this value to the signin call:

```typescript
    supabase.auth.signIn(credentials, {
      redirectTo: 'mydomain.com/app/route-that-needs-auth'
    });
```

...but the above does not work unless I add `/route-that-needs-auth` to my `Additional Redirect URLs` config in supabase dashboard. This obviously isnt scalable, because I would like to be able to redirect the user back to potentially any route in the application.

Another approach I tried was redirecting to a `/redirect?destination=/app/route-that-needs-auth`, but it seems to fail for the same reason (the addition of the query param does not match with the `Additional Redirect URLs`).

Is what I'm looking for achievable?

**Comment by shoxadomokai** [comment]
To solve your issue; I've stored the redirect route needed in a cookie session with the key "redirect" and in the auth handler for my site I check if there's a "redirect" key in my cookie session. If there is I redirect the user to the path and clear the key. 

The bit I am struggling with in Additional Redirect URLs is only the very first URL in the comma-separated list is accepted/working.
